File: fsharp/space-age/SpaceAge.fs
module SpaceAge

type Planet = 
    | Earth
    | Mercury
    | Venus
    | Mars
    | Jupiter
    | Saturn
    | Uranus
    | Neptune

let age (planet: Planet) (seconds: int64): float =
    let earthYears = (float seconds) / 31557600.0
    match planet with
    | Earth -> earthYears
    | Mercury -> earthYears / 0.2408467
    | Venus -> earthYears / 0.61519726
    | Mars -> earthYears / 1.8808158
    | Jupiter -> earthYears / 11.862615
    | Saturn -> earthYears / 29.447498
    | Uranus -> earthYears / 84.016846
    | Neptune -> earthYears / 164.79132

File: go/space-age/space_age.go
package space

// Planet is a planet in our solar system.
type Planet string

const secondsInEarthYear = 31557600.0

var orbitalPeriods = map[Planet]float64{
	"Mercury": 0.2408467,
	"Venus":   0.61519726,
	"Earth":   1.0,
	"Mars":    1.8808158,
	"Jupiter": 11.862615,
	"Saturn":  29.447498,
	"Uranus":  84.016846,
	"Neptune": 164.79132,
}

// Age calculates the age on the given planet based on the given age in seconds.
func Age(seconds float64, planet Planet) float64 {
	earthYears := seconds / secondsInEarthYear

	if period, ok := orbitalPeriods[planet]; ok {
		return earthYears / period
	}

	return 0
}
